atomic_reactor/plugins/pre_flatpak_update_dockerfile.py

```diff
@@ -0,0 +1,257 @@
+                                                            module_spec.version,
+                                                            mmd, rpms)
+        return resolved_modules
+
+    def _build_compose_info(self, modules):
+        source_spec = get_flatpak_source_spec(self.workflow)
+        assert source_spec is not None  # flatpak_create_dockerfile must be run first
+        main_module = ModuleSpec.from_str(source_spec)
+
+        resolved_modules = self._resolve_modules(modules)
+
+        main_module_info = resolved_modules[main_module.name]
+        assert main_module_info.stream == main_module.stream
+        if main_module.version is not None:
+            assert main_module_info.version == main_module.version
+
+        return ComposeInfo(source_spec=source_spec,
+                           main_module=main_module_info,
+                           modules=resolved_modules)
+
+    def _load_compose_info(self):
+        source_spec = get_flatpak_source_spec(self.workflow)
+        assert source_spec is not None  # flatpak_create_dockerfile must be run first
+        main_module = ModuleSpec.from_str(source_spec)
+
+        resolve_comp_result = self.workflow.prebuild_results.get(PLUGIN_RESOLVE_COMPOSES_KEY)
+        if resolve_comp_result:
+            # In the orchestrator, we can get the compose info directly from
+            # the resolve_composes plugin
+            composes = resolve_comp_result['composes']
+        else:
+            # But in a worker, resolve_composes doesn't run, so we have
+            # to load the compose info ourselves
+            assert self.compose_ids
+
+            composes = self._load_composes()
+
+        for compose_info in composes:
+            if compose_info['source_type'] != SOURCE_TYPE_MODULE:
+                continue
+
+            modules = [ModuleSpec.from_str(s) for s in compose_info['source'].split()]
+            for module in modules:
+                if module.name == main_module.name and module.stream == main_module.stream:
+                    return self._build_compose_info(modules)
+
+        self.log.debug('Compose info: %s', composes)
+        raise RuntimeError("Can't find main module %s in compose result" % source_spec)
+
+    def _load_source(self):
+        flatpak_yaml = self.workflow.source.config.flatpak
+
+        compose_info = get_flatpak_compose_info(self.workflow)
+
+        module_spec = ModuleSpec.from_str(compose_info.source_spec)
+
+        return FlatpakSourceInfo(flatpak_yaml,
+                                 compose_info.modules,
+                                 compose_info.main_module,
+                                 module_spec.profile)
+
+    def run(self):
+        """
+        run the plugin
+        """
+
+        compose_info = self._load_compose_info()
+        set_flatpak_compose_info(self.workflow, compose_info)
+
+        source = self._load_source()
+        set_flatpak_source_info(self.workflow, source)
```

Comment:
since this and set_flatpak_compose_info are done unconditionally, is there any reason to not set them in `_load_*()` ?

Reply:
You could - but then I don't like having _load_source() both set the source info and have a return value, so I switched to
```python
        self._load_source()
        source = get_flatpak_source_info(self.workflow, source)
```
(which is more like the other places in the code where the source_info is retrieved)